extension/src/tasks/taskUtil.ts

```diff
@@ -205,6 +206,72 @@ export function createTaskFromDefinition(
   return task;
 }
 
+export function resolveTaskFromDefinition(
```

Comment:
Here just copy the main logics from `createTaskFromDefinition` and use `CustomExecution` to resolve the VSCode variables